### DISCUSSION TITLE
API rules cheat sheet

### DISCUSSION THREAD

**Discussion by FrenchFryInteractive** [pocketbase/pocketbase#3162] [Q&A]
Does anyone have cheat sheet for API rules, like how to set rules that only the creator can delete or how do I use collection data in my api rules. I am all kind of confused about API rules.

**Comment by ganigeorgiev** [comment]
I don't think this can be done universally as it will depend on your collections configuration.

Did you check the short overview at https://pocketbase.io/docs/api-rules-and-filters/?

Could you elaborate what exactly confused you about the API rules so that I can try to improve the documentation?

---

In general if a collection has for example a **non-empty "owner" user relation** field, then you can apply the following rules:

1. List/Search API rule - allow listing only the records that the current authenticated user owns
    ```js
    owner = @request.auth.id
    ```

2. View API rule - allow only owners
    ```js
    owner = @request.auth.id
    ```

3. Create API rule - require the submitted "owner" field to match the current authenticated user
    ```js
    @request.data.owner = @request.auth.id
    ```
    > _Or alternatively, because the create API rule is applied after a dry submit, the below should also work:_
    > ```js
    > owner = @request.auth.id
    > ```

4. Update API rule - allow only owners AND prevent reassignment of the "owner" field
    ```js
    owner = @request.auth.id &&
    @request.data.owner:isset = false
    ```

5. Delete API rule - allow only owners
    ```js
    owner = @request.auth.id
    ```
